contrib/CONTRIBUTING.md

```diff
@@ -0,0 +1,232 @@
+# Contributing to Bittensor
+
+The following is a set of guidelines for contributing to Bittensor, which are hosted in the [Opentensor Organization](https://github.com/opentensor) on GitHub. These are mostly guidelines, not rules. Use your best judgment, and feel free to propose changes to this document in a pull request.
+
+## Table Of Contents
+
+1. [I don't want to read this whole thing, I just have a question!!!](#i-dont-want-to-read-this-whole-thing-i-just-have-a-question)
+1. [What should I know before I get started?](#what-should-i-know-before-i-get-started)
+1. [How Can I Contribute?](#how-can-i-contribute)
+   1. [Code Contribution General Guideline](#code-contribution-general-guidelines)
+   1. [Pull Request Philosophy](#pull-request-philosophy)
+   1. [Pull Request Process](#pull-request-process)
+   1. [Testing](#testing)
+   1. [Addressing Feedback](#addressing-feedback)
+   1. [Squashing Commits](#squashing-commits)
+   1. [Refactoring](#refactoring)
+ 1. [Reporting Bugs](#reporting-bugs)
+ 1. [Suggesting Features](#suggesting-enhancements)
+
+
+## I don't want to read this whole thing I just have a question!
+
+> **Note:** Please don't file an issue to ask a question. You'll get faster results by using the resources below.
+
+We have an official Discord server where the community chimes in with helpful advice if you have questions. 
+This is the fastest way to get an answer and the core development team is active on Discord.
+
+* [Official Bittensor Discord](https://discord.gg/7wvFuPJZgq)
+
+## What should I know before I get started?
+Bittensor is still in the Alpha stages, and as such you will likely run into some problems in deploying your model or installing Bittensor itself. If you run into an issue or end up resolving an issue yourself, feel free to create a pull request with a fix or with a fix to the documentation. The documentation repository can be found [here](https://github.com/opentensor/docs). 
+
+Additionally, note that the core implementation of Bittensor consists of two separate repositories: [The core Bittensor code](https://github.com/opentensor/bittensor) and the Bittensor Blockchain [subtensor](https://github.com/opentensor/subtensor).
+
+Supplemental, yet necessary repositories are [openvalidators](https://github.com/opentensor/validators) and [openminers](https://github.com/opentensor/miners).
+
+## How Can I Contribute?
+
+You can contribute to Bittensor in one of three main ways (as well as many others):
+1. [Bug](#reporting-bugs) reporting and fixes
```

Comment:
The current bug and feature request templates should be converted to [forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms). The reason for this is that users in general tend to take the path of least resistance and delete the whole template and write down 3 words instead. The form fields can be marked as required and force a structured format for the report.

Reply:
+1, that would be a very nice thing to have!